smurfs/src/reducers/index.js

```diff
@@ -1,7 +1,50 @@
+import { 
+  FETCH_SMURFS, 
+  SUCCESS, 
+  FAILURE, 
+  POST_SMURF, 
+  DELETE_SMURF, 
+  PUT_SMURF,
+} from '../actions';
+
 /*
   Be sure to import in all of the action types from `../actions`
 */
 
+const initialState = {
+  smurfs: [],
+  fetchingSmurfs: false,
+  error: null,
+  editing: false,
+}
+
+export const Reducer = (state = initialState, action) => {
+  switch(action.type){
+    case FETCH_SMURFS:
+      return {
+        ...state,
+        fetchingSmurfs: true,
+      }
+
+    case SUCCESS:
+      return {
+        ...state,
+        fetchingSmurfs: false,
+        smurfs: [...action.payload]
+      }
+
+    case FAILURE:
+      return {
+        ...state,
+        fetchingSmurfs: false,
+        error: action.payload,
+      }
+    default:
+      return state;
+  }
+}
+
```

Comment:
Easy way to handle your cases, looks really clean and straightforward. I'd recommend if you have the time to work on the stretch goals as well and push yourself a bit more, I don't think it would be a big deal for you since you clearly know the content.
Great job in your SC, I'm looking forward to see what you on do on Build Week